### PR TITLE
feat: handle unsupported version exception, closes MISC-374

### PR DESCRIPTION
**Pull request by tpetillot**
chore: centralize plugin exception handling accross tasks
---

**Motivation:**
Recovering from error is always the same

**Modification:**
Add RecoverEnterprisePluginException to extend in order to get recovering method

 
feat: handle unsupported version exception, closes MISC-374
---

**Motivation:**
New commons version throw `UnsupportedJavaVersionException` when java version isn't supported by Gatling Enterprise CLoud

**Modification:**
Log exception message, along java option to target a compatible bytecode.

### CODE REVIEW

**Review by slandelle — DISMISSED** [review]

File: src/main/groovy/io/gatling/gradle/GatlingEnterpriseUploadTask.groovy [hunk 1]
```diff
@@ -11,18 +11,20 @@ class GatlingEnterpriseUploadTask extends DefaultTask {
...
-                logger.lifecycle("Uploading package with packageId " + gatling.enterprise.packageId)
-                it.uploadPackage(gatling.enterprise.packageId, inputs.files.singleFile)
-            } else if (gatling.enterprise.simulationId) {
-                logger.lifecycle("Uploading package belonging to the simulation " + gatling.enterprise.simulationId)
-                it.uploadPackageWithSimulationId(gatling.enterprise.simulationId, inputs.files.singleFile)
-            } else {
-                throw new InvalidUserDataException("You need to either configure gatling.enterprise.packageId (or pass it with '-Dgatling.enterprise.packageId=<PACKAGE_ID>') " +
-                    "or gatling.enterprise.simulation (or pass it with '-Dgatling.enterprise.simulationId=<SIMULATION_ID>') to upload a package." +
-                    "Please see https://gatling.io/docs/gatling/reference/current/extensions/gradle_plugin/#working-with-gatling-enterprise-cloud for more information.")
+        RecoverEnterprisePluginException.handle(this, { ->
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/GatlingEnterpriseUploadTask.groovy:14]:
```suggestion
        RecoverEnterprisePluginException.handle(this, {
```

File: src/main/groovy/io/gatling/gradle/GatlingEnterpriseStartTask.groovy [hunk 2]
```diff
@@ -27,75 +22,28 @@ class GatlingEnterpriseStartTask extends DefaultTask {
...
-                    gatling.enterprise.initInteractiveEnterprisePlugin(version, logger)
+        final EnterprisePlugin enterprisePlugin =
+            gatling.enterprise.batchMode ?
+                gatling.enterprise.initBatchEnterprisePlugin(version, logger) :
+                gatling.enterprise.initInteractiveEnterprisePlugin(version, logger)
 
-            def simulationStartResult = gatling.enterprise.simulationId ?
-                enterprisePlugin.uploadPackageAndStartSimulation(simulationId, systemProperties, simulationClass, file) :
-                enterprisePlugin.createAndStartSimulation(teamId, groupId, artifactId, simulationClass, packageId, systemProperties, file)
+        final SimulationStartResult simulationStartResult = RecoverEnterprisePluginException.handle(this, {
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/GatlingEnterpriseStartTask.groovy:30]:
```suggestion
        final SimulationStartResult simulationStartResult = RecoverEnterprisePluginException.handle({
```

File: src/main/groovy/io/gatling/gradle/RecoverEnterprisePluginException.groovy [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.gradle.api.tasks.TaskExecutionException
+
+class RecoverEnterprisePluginException {
+
+    /**
+     * @param f Closure defined inside a Task
+     * @param logger used to report EnterprisePluginException
+     * @return closure result
+     */
+    static <R> R handle(Closure<R> f, Logger logger) {
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/RecoverEnterprisePluginException.groovy:17]: Groovy convention:
When a method has multiple arguments and one and only one is a Closure, put it last.

```suggestion
    static <R> R handle(Logger logger, Closure<R> f) {
```

That allows a syntax similar to currying:

```groovy
  RecoverEnterprisePluginException.handle(logger) {
    // your closure content here
  }
```

File: src/main/groovy/io/gatling/gradle/GatlingEnterpriseUploadTask.groovy [hunk 4]
```diff
@@ -11,18 +11,20 @@ class GatlingEnterpriseUploadTask extends DefaultTask {
...
-                logger.lifecycle("Uploading package with packageId " + gatling.enterprise.packageId)
-                it.uploadPackage(gatling.enterprise.packageId, inputs.files.singleFile)
-            } else if (gatling.enterprise.simulationId) {
-                logger.lifecycle("Uploading package belonging to the simulation " + gatling.enterprise.simulationId)
-                it.uploadPackageWithSimulationId(gatling.enterprise.simulationId, inputs.files.singleFile)
-            } else {
-                throw new InvalidUserDataException("You need to either configure gatling.enterprise.packageId (or pass it with '-Dgatling.enterprise.packageId=<PACKAGE_ID>') " +
-                    "or gatling.enterprise.simulation (or pass it with '-Dgatling.enterprise.simulationId=<SIMULATION_ID>') to upload a package." +
-                    "Please see https://gatling.io/docs/gatling/reference/current/extensions/gradle_plugin/#working-with-gatling-enterprise-cloud for more information.")
+        RecoverEnterprisePluginException.handle(this, {
```
Comment (Isammoc) [inline src/main/groovy/io/gatling/gradle/GatlingEnterpriseUploadTask.groovy:14]:
```suggestion
        RecoverEnterprisePluginException.handle({
```

**Review by Isammoc — APPROVED** [review]

**Pull request merged**